### PR TITLE
timeout, user-agent, core features tests

### PR DESCRIPTION
**Pull request by Zerryth**
* more tests for core qna features
* switched from requests library to aiohttp as http handler, to allow for queries to qna service to be asynchronous
* timeout and user-agent added to qna

### CODE REVIEW

File: libraries/botbuilder-ai/tests/qna/test_qna.py [hunk 1]
```diff
@@ -3,21 +3,35 @@
...
+        self.sent: List[Activity] = list()
+
+        self.on_send_activities(self.capture_sent_activities)
+
+    async def capture_sent_activities(self, context: TurnContext, activities, next):
+        self.sent += activities
+        context.responded = True
 
 class QnaApplicationTest(aiounittest.AsyncTestCase):
     _knowledge_base_id: str = 'a090f9f3-2f8e-41d1-a581-4f7a49269a0c'
```
Comment (daveta) [inline libraries/botbuilder-ai/tests/qna/test_qna.py]: > 'a090f9f3-2f8e-41d1-a581-4f7a49269a0c' [](start = 29, length = 39)

You should remove these..

**Comment by daveta** [comment]
>     "aiounittest>=1.1.0",

Might put this into separate TEST_REQUIRES

---
Refers to: libraries/botbuilder-ai/setup.py:8 in 90de2da. [](commit_id = 90de2da6f08d8b5ce94fca696b23323ff4d3da0e, deletion_comment = False)

**Comment by daveta** [comment]
>     keywords=["BotBuilderDialogs", "bots", "ai", "botframework", "botbuilder"],

Probably want to change this..

---
Refers to: libraries/botbuilder-ai/setup.py:29 in 90de2da. [](commit_id = 90de2da6f08d8b5ce94fca696b23323ff4d3da0e, deletion_comment = False)

**Comment by daveta** [comment]
>     packages=["botbuilder.ai"],

I'm assuming we want to add botbuilder.ai.qna

---
Refers to: libraries/botbuilder-ai/setup.py:32 in 90de2da. [](commit_id = 90de2da6f08d8b5ce94fca696b23323ff4d3da0e, deletion_comment = False)

**Comment by daveta** [comment]
Nice!

File: libraries/botbuilder-ai/tests/qna/test_qna.py (hunk 1 above)
Comment (Zerryth) [reply libraries/botbuilder-ai/tests/qna/test_qna.py]: ...AHH!

will do

File: libraries/botbuilder-ai/setup.py [hunk 2]
```diff
@@ -25,11 +28,12 @@
...
     author=package_info["__author__"],
     description=package_info["__description__"],
-    keywords=["BotBuilderDialogs", "bots", "ai", "botframework", "botbuilder"],
+    keywords="botbuilder-ai LUIS QnAMaker bots ai botframework botbuilder",
     long_description=package_info["__summary__"],
     license=package_info["__license__"],
-    packages=["botbuilder.ai"],
-    install_requires=REQUIRES,
+    packages=["botbuilder.ai", "botbuilder.ai.qna", "botbuilder.ai.luis"],
+    install_requires=REQUIRES + TESTS_REQUIRES,
```
Comment (daveta) [inline libraries/botbuilder-ai/setup.py]: Can you remove "+ TEST_REQUIRES"

**Pull request merged**